lib/datasets/aozora-bunko.rb

```diff
@@ -0,0 +1,164 @@
+      # 人物ID, 姓,名,姓読み,名読み,姓読みソート用,名読みソート用,姓ローマ字,名ローマ字,役割フラグ,生年月日,没年月日,人物著作権フラグ,
+      :person_id,
+      :person_family_name,
+      :person_first_name,
+      :person_family_name_reading,
+      :person_first_name_reading,
+      :person_family_name_collation,
+      :person_first_name_collation,
+      :person_family_name_roman,
+      :person_first_name_roman,
+      :person_type,
+      :person_birthday,
+      :person_death_day,
+      :copyright_for_person,
+      # 底本名1,底本出版社名1,底本初版発行年1,入力に使用した版1,校正に使用した版1,底本の親本名1,底本の親本出版社名1,底本の親本初版発行年1,
+      :original_book_name1,
+      :original_book_publisher_name1,
+      :original_book_first_published_date1,
+      :used_version_for_registration1,
+      :used_version_for_proofreading1,
+      :base_of_original_book_name1,
+      :base_of_original_book_publisher_name1,
+      :base_of_original_book_first_published_date1,
+      # 底本名2,底本出版社名2,底本初版発行年2,入力に使用した版2,校正に使用した版2,底本の親本名2,底本の親本出版社名2,底本の親本初版発行年2,
+      :original_book_name2,
+      :original_book_publisher_name2,
+      :original_book_first_published_date2,
+      :used_version_for_registration2,
+      :used_version_for_proofreading2,
+      :base_of_original_book_name2,
+      :base_of_original_book_publisher_name2,
+      :base_of_original_book_first_published_date2,
+      # 入力者,校正者,
+      :registered_person_name,
+      :proofreader_name,
+      # テキストファイルURL,テキストファイル最終更新日,テキストファイル符号化方式,テキストファイル文字集合,テキストファイル修正回数,
+      :text_file_url,
+      :latest_text_file_updated_date,
+      :text_file_character_encoding,
+      :text_character_set,
+      :text_file_updating_count,
+      # XHTML/HTMLファイルURL,XHTML/HTMLファイル最終更新日,XHTML/HTMLファイル符号化方式,XHTML/HTMLファイル文字集合,XHTML/HTMLファイル修正回数
+      :html_file_url,
+      :latest_html_file_updated_date,
+      :html_file_character_encoding,
+      :html_character_set,
+      :html_file_updating_count
+    )
+
+    class Record
+      def text
+        return @text unless @text.nil?
+        return '' if text_file_url.nil? || text_file_url.empty?
+
+        # when url is not zip file, it needs to open web page by brower and has to download
+        # e.g. https://mega.nz/file/6tMxgAjZ#PglDDyJL0syRhnULqK0qhTMC7cktsgqwObj5fY_knpE
+        return '' unless text_file_url.end_with?('.zip')
+
+        downloader = Datasets::Downloader.new(text_file_url)
+        downloader.download(output_path)
+
+        Zip::File.open(output_path) do |zip_file|
+          zip_file.each do |entry|
+            next unless entry.file?
+
+            entry.get_input_stream do |stream|
+              @text = stream.read.encode(Encoding::UTF_8, Encoding::SHIFT_JIS)
+            end
+          end
+        end
+
+        clear_cache!
+
+        @text
+      end
+
+      private
+
+      def output_path
+        cache_path.base_dir + text_file_url.split('/').last
+      end
+
+      def clear_cache!
+        cache_path.remove
+      end
+
+      def cache_path
+        @cache_path ||= CachePath.new("aozora-bunko/#{title_id}")
```

Comment:
It seems that `title_id` may be duplicated. How about `person_id` too?

```suggestion
        @cache_path ||= CachePath.new("aozora-bunko-#{title_id}-#{person_id}")
```